analytics/pubstack/pubstack_module_test.go

```diff
@@ -57,130 +55,158 @@ func loadJsonFromFile() (*analytics.AuctionObject, error) {
-	assert.Nil(t, err)
-
-	// Default Configuration
-	pubstack, ok := module.(*PubstackModule)
-	assert.Equal(t, ok, true) //PBSAnalyticsModule is also a PubstackModule
-	assert.Equal(t, len(pubstack.cfg.Features), 5)
-	assert.Equal(t, pubstack.cfg.Features[auction], false)
-	assert.Equal(t, pubstack.cfg.Features[video], false)
-	assert.Equal(t, pubstack.cfg.Features[amp], false)
-	assert.Equal(t, pubstack.cfg.Features[setUID], false)
-	assert.Equal(t, pubstack.cfg.Features[cookieSync], false)
-
-	assert.Equal(t, len(pubstack.eventChannels), 0)
-
-	// Process Auction Event
-	counter := 0
-	send := func(_ []byte) error {
-		counter++
-		return nil
-	}
-	mockedEvent, err := loadJsonFromFile()
-	if err != nil {
-		t.Fail()
+func TestPubstackModuleErrors(t *testing.T) {
+	tests := []struct {
+		giveRefreshDelay string
+		giveMaxByteSize  string
+		giveMaxTime      string
+	}{
+		{
+			giveRefreshDelay: "1z",
+			giveMaxByteSize:  "90MB",
+			giveMaxTime:      "15m",
+		},
+		{
+			giveRefreshDelay: "1h",
+			giveMaxByteSize:  "90z",
+			giveMaxTime:      "15m",
+		},
+		{
+			giveRefreshDelay: "1z",
+			giveMaxByteSize:  "90MB",
+			giveMaxTime:      "15z",
+		},
 	}
 
-	pubstack.eventChannels[auction] = eventchannel.NewEventChannel(send, 2000, 1, 10*time.Second)
-	pubstack.eventChannels[video] = eventchannel.NewEventChannel(send, 2000, 1, 10*time.Second)
-	pubstack.eventChannels[amp] = eventchannel.NewEventChannel(send, 2000, 1, 10*time.Second)
-	pubstack.eventChannels[setUID] = eventchannel.NewEventChannel(send, 2000, 1, 10*time.Second)
-	pubstack.eventChannels[cookieSync] = eventchannel.NewEventChannel(send, 2000, 1, 10*time.Second)
-
-	pubstack.LogAuctionObject(mockedEvent)
-	pubstack.LogAmpObject(&analytics.AmpObject{
-		Status: http.StatusOK,
-	})
-	pubstack.LogCookieSyncObject(&analytics.CookieSyncObject{
-		Status: http.StatusOK,
-	})
-	pubstack.LogVideoObject(&analytics.VideoObject{
-		Status: http.StatusOK,
-	})
-	pubstack.LogSetUIDObject(&analytics.SetUIDObject{
-		Status: http.StatusOK,
-	})
-
-	pubstack.closeAllEventChannels()
-	time.Sleep(10 * time.Millisecond) // process channel
-	assert.Equal(t, counter, 0)
-
-	// Hot-Reload config
-	newFeatures := make(map[string]bool)
-	newFeatures[auction] = true
-	newFeatures[video] = true
-	newFeatures[amp] = true
-	newFeatures[cookieSync] = true
-	newFeatures[setUID] = true
-
-	remoteConfig = &Configuration{
-		ScopeID:  "new-scope",
-		Endpoint: "new-endpoint",
-		Features: newFeatures,
+	for _, tt := range tests {
+		server := httptest.NewServer(http.HandlerFunc(func(res http.ResponseWriter, req *http.Request) {
+			defer req.Body.Close()
+			data, _ := json.Marshal(&Configuration{})
+			res.Write(data)
+		}))
+		client := server.Client()
+
+		defer server.Close()
+
+		_, err := NewPubstackModule(client, "scope", "http://example.com", tt.giveRefreshDelay, 100, tt.giveMaxByteSize, tt.giveMaxTime)
+		assert.NotNil(t, err) // should raise an error since  we can't parse args // configRefreshDelay
```

Comment:
Why not? I'm guessing because "z" isn't a valid label? Consider adding a comment in the test cases to clarify, or better yet a description per test case. Also, please include the test description in this assert statement so we know which case failed.